graphql_compiler/tests/test_macro_validation.py

```diff
@@ -208,7 +208,29 @@ def test_macro_edge_extra_args(self):
         }
 
         macro_registry = create_macro_registry()
-        with self.assertRaises(GraphQLInvalidMacroError):
+        with self.assertRaises(GraphQLInvalidArgumentError):
+            register_macro_edge(macro_registry, self.schema, query,
+                                args, self.type_equivalence_hints)
+
+    def test_macro_edge_incorrect_arg_types(self):
+        query = '''{
+            Animal @macro_edge_definition(name: "out_Animal_GrandparentOf") {
+                net_worth @filter(op_name: "=", value: ["$net_worth"])
+                color @filter(op_name: "=", value: ["$color"])
+                out_Animal_ParentOf {
+                    out_Animal_ParentOf @macro_edge_target {
+                        uuid
+                    }
+                }
+            }
+        }'''
+        args = {
+            'net_worth': 'five_cows',
```

Comment:
Suggestion: instead of `five_cows`, let's make this `incorrect_net_worth_type`? And maybe add a quick comment in the same line `# valid types are x and y`